data/trie/trieStorageManager.go

```diff
@@ -275,32 +275,44 @@ func (tsm *trieStorageManager) snapshot(msh marshal.Marshalizer, hsh hashing.Has
 		tr, err := newSnapshotTrie(tsm.db, msh, hsh, snapshot.rootHash)
 		if err != nil {
 			tsm.storageOperationMutex.Unlock()
+			isSnapshotsBufferEmpty, keys = tsm.isSnapshotsBufferEmpty()
+
 			log.Error("trie storage manager: newSnapshotTrie", "error", err.Error())
-			return
+			break
 		}
 		db := tsm.getSnapshotDb(snapshot.newDb)
 
 		tsm.storageOperationMutex.Unlock()
 
 		err = tr.root.commit(true, 0, tsm.db, db)
 		if err != nil {
+			isSnapshotsBufferEmpty, keys = tsm.isSnapshotsBufferEmpty()
```

Comment:
_, keys = ... ?

Reply:
No, every time the for loop breaks, isSnapshotsBufferEmpty must be recomputed, otherwise there will be a deadlock.

Reply:
But you have a break just after the next line, there would be no more loops